docs/tips/19-dangerously-set-inner-html.md

```diff
@@ -17,8 +17,8 @@ function createMarkup() { return {__html: 'First &middot; Second'}; };
 <div dangerouslySetInnerHTML={createMarkup()} />
 ```
 
-The point being that if you unintentionally do say `<div dangerouslySetInnerHTML={getUsername()} />` it will not be rendered because `getUsername()` would return a plain `string` and not a `{__html: ''}` object.  The intent behind the `{__html:...}` syntax is that it be considered a "type/taint" of sorts.  Sanitized data can be returned from a function using this wrapper object, and this marked data can subsequently be passed into `dangerouslySetInnerHTML`.  For this reason, we recommend against writing code of the form `<div dangerouslySetInnerHTML={{'{{'}}__html: getMarkup()}} />`.
+The intent behind the `{__html:...}` syntax is to ensure that the developer has taken the necessary steps to sanitize the data to be rendered. The point being that if you unintentionally say `<div dangerouslySetInnerHTML={getUsername()} />`, it will not be rendered because `getUsername()` would return a plain `string` and not a `{__html: ''}` object. This prevents unsanitized data from being rendered and acts as a form of [taint checking](https://en.wikipedia.org/wiki/Taint_checking). Sanitized data can be returned from a function using this wrapper object, and this marked data can subsequently be passed into `dangerouslySetInnerHTML`.  For this reason, we recommend against writing code of the form `<div dangerouslySetInnerHTML={{'{{'}}__html: getMarkup()}} />`. 
 
-This functionality is mainly provided for cooperation with DOM string manipulation libraries, so the HTML provided must be well-formed (ie., pass XML validation).
+> Note: The HTML provided within the `{__html: ...}` object must be well-formed (ie., pass XML validation) in order to cooperate with DOM string manipulation libraries.
```

Comment:
The old version was correct.  The `dangerouslySetInnerHTML` functionality is provided in order to cooperate with DOM string manipulation libraries.  The XML should be well-formed, and should be coming from a string manipulation library.